core/trino-main/src/main/java/io/trino/metadata/MetadataListing.java

```diff
@@ -98,6 +98,10 @@ private MetadataListing() {}
         Set<SchemaTableName> tableNames = metadata.listTables(session, prefix).stream()
                 .map(QualifiedObjectName::asSchemaTableName)
                 .collect(toImmutableSet());
+
+        // Table listing operation only involves getting table names, but not any metadata. So redirected tables are not
+        // handled any differently. The target table or catalog are not involved. Thus the following filter is only called
+        // for the source catalog on source table names.
```

Comment:
So we don't filter on target table name? I though this redirection is logically similar to symlink so we perform every action (e.g access control checks) on targets.